### DISCUSSION TITLE
Safari touch can't paste content in browser version without typing a character first?

### DISCUSSION THREAD

**Discussion by LorenAmelang** [TriliumNext/Trilium#2783] [General]
I know this is a complex and unsupported situation! But it is a huge portion of my workflow...  I connect to my Raspberry PI server from my iPad (12.9 5gen, 15.4.1) Safari browser, create a child note, and try to paste some content. Seems the only way to get Safari to show the Paste option when I tap in the note area is to type a character where I want to paste and then tap after it. Later I can delete the extra character. 

Just tried this in Vivaldi browser on Windows 10, and a single touch shows the Paste option anywhere in the note, so maybe it has nothing to do with Trilium. 

But on the other hand, if the clipboard is occupied, Safari generally offers to Paste anywhere characters can be added. What could be different about a Trilium note line with a visible cursor but no characters, and one with a character on the line already? Safari apparently sees some difference! 

Maybe there is some way to add a Paste command to the side Panel?

**Comment by zadam** [comment]
Hi, is it reproducible on vanilla CKEditor? https://ckeditor.com/ckeditor-5/demo/

**Reply by LorenAmelang** [reply]
No, I couldn't duplicate the need to type a character before seeing the paste prompt. In the test version I sometimes get the cursor and paste prompt on the first tap on a new line. Sometimes only the cursor appears in the new location, but in those cases another tap on that line _always_ gets the paste prompt. Where I tap (left or right of the left margin, or way to the right or way down where Trilium would ignore the tap) doesn't seem to matter. 

The format of the blank line doesn't seem to matter, nor does what's above or below it. In both cases press and hold on a new line pops up the draggable single vertical line selection cursor and tiny preview above the line, but of course there is nothing to select. 

Yes, there are other disadvantages to touch-only use. Like if I tap the paragraph symbol to the left of the line in Trilium, it disappears and the on-screen keyboard is hidden. Seems no way to change formatting. The CKEditor test shows the format menu in its header, and it works - maybe you could have an option to do that? 

But fancy formatting can be done later. Pasting the raw content into a note from mobile is more important! At least now I know it works if I type a character first!

![CKEditor 5 Demo - Rich Text Editor ready to use](https://user-images.githubusercontent.com/1297090/162830120-6ba53a01-a8e5-4ec4-9d51-da5231997ab7.png)
